apps/src/applab/AppLabCrosshairOverlay.jsx

```diff
@@ -0,0 +1,40 @@
+/** @file App Lab-specific Crosshair Overlay */
+
+import CrosshairOverlay from '../templates/CrosshairOverlay';
+import { scaledDropPoint } from './gridUtils';
+
+const AppLabTooltipOverlay = React.createClass({
+  propTypes: {
+    // width, height, mouseX and mouseY are given in app-space, not screen-space
+    width: React.PropTypes.number,
+    height: React.PropTypes.number,
+    mouseX: React.PropTypes.number,
+    mouseY: React.PropTypes.number
+  },
+
+  render() {
+    var mouseX = this.props.mouseX;
+    var mouseY = this.props.mouseY;
+
+    // Modify passed props if we're in a 'dragging' mode.
+    var draggingElement = $(".ui-draggable-dragging");
+    this.isDragging_ = !!draggingElement.length;
+    if (this.isDragging_) {
+      // If we're dragging an element, use its current drop position
+      // (top left of the dragged element)
+      var point = scaledDropPoint(draggingElement);
+      mouseX = point.left;
+      mouseY = point.top;
+    }
```

Comment:
if you want to use fancy es6 syntax, you could write

`var {left: mouseX, top: mouseY} = scaledDropPoint(draggingElement);`

Depending of course on whether you consider that more readable (I'm not sure it is)